carp.studies.core/src/commonMain/kotlin/dk/cachet/carp/studies/application/RecruitmentServiceHost.kt

```diff
@@ -160,21 +168,86 @@ class RecruitmentServiceHost(
     }
 
     /**
-     * Get the status of all deployed participant groups in the study with the specified [studyId].
+     * Create a new participant [group] of previously added participants for the study with the given [studyId],
+     * and an optional [name] representing this group, but do not yet send out invitations.
+     * This is used to create a group of participants which can be deployed at a later time.
+     *
+     * As long as no final study protocol is locked in for the study, a participant group can't be created
+     * since participant roles to which participants need to be assigned are unknown.
+     * @throws IllegalArgumentException when:
+     *  - a study with [studyId] does not exist
+     *  - an existing participant group with [groupId] already exists
+     *  - any of the participant roles specified in [group] does not exist
+     * @throws IllegalStateException when the study is not yet ready for deployment.
+     */
+    override suspend fun createParticipantGroup(
+        groupId: UUID,
+        group: Set<AssignedParticipantRoles>,
+        studyId: UUID,
+        name: String?
+    ): ParticipantGroupStatus
+    {
+        val recruitment = getRecruitmentOrThrow( studyId )
+        require( participantRepository.getRecruitmentWithParticipantGroup( groupId ) == null )
+            { "Participant group with ID \"$groupId\" already exists." }
```

Comment:
@yuanchen233 Note that I changed this to check for _any_ `groupId`, instead of an existing `groupId` within the `recruitment`.